### DISCUSSION TITLE
Where exactly are osu! resources stored?

### DISCUSSION THREAD

**Discussion by 0xdeadbeer** [ppy/osu#16255] [Q&A] [answered]
I'm somewhat new to the osu!'s codebase and don't know a lot when it comes to the osu framework. 
I was going through the source code trying to find a bug/challenge and noticed the `\osu.Game\Graphics\Cursor\MenuCursor.cs` file.. the name was interesting so I decided to take a quick look at it.. 
I understood pretty much everything except the following line: 
`Texture = textures.Get(@"Cursor/menu-cursor"),`
I do know that it will return the texture of the menu cursor and save it as a "property" for the sprite "element".. 
But.. **where is the program getting the images from? I can ask the same for other resources like tracks, icons, etc..**

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu-resources

You can clone this and use a local checkout following these instructions: https://github.com/ppy/osu-resources/wiki/Testing-local-resources-checkout-with-other-projects

**Reply by 0xdeadbeer** [reply]
Local? So the ones that I'm using right now are stored online?

**Reply by bdach** [reply]
they're not "stored" online, the files from the above repository are packaged as embedded resources in a dll that is later distributed via a nuget package, so you would not be easily able to change any of the resources without a local checkout (which is replacing the nuget package reference with a local project reference).

**Reply by 0xdeadbeer** [reply]
Got that, thanks for reaching out!
